tests/tools.vitruv.applications.pcmumlclass.tests/src/tools/vitruv/applications/pcmumlclass/tests/NewPcmUmlClassApplicationTest.xtend

```diff
@@ -0,0 +1,140 @@
+	@Accessors(PROTECTED_GETTER)
+	static val MODEL_FOLDER_NAME = "model"
+
+	static val PACKAGE_NAME = "rootpackage"
+	static val PACKAGE_NAME_FIRST_UPPER = "Rootpackage"
+	static val CONTRACTS_PACKAGE = "contracts"
+
+	protected var extension PcmUmlclassViewFactory viewFactory
+	protected var PcmUmlClassApplicationTestHelper helper
+	protected var ResourceSet testResourceSet
+
+	override protected getChangePropagationSpecifications() {
+		return #[
+			new CombinedPcmToUmlClassReactionsChangePropagationSpecification,
+			new CombinedUmlClassToPcmReactionsChangePropagationSpecification
+		]
+	}
+
+	protected def getTestResource(URI uri) {
+		return testResourceSet.getResource(uri, true)
+	}
+
+	protected def void createUmlModel((Model)=>void modelInitialization) {
+		changeUmlView [
+			val umlModel = UMLFactory.eINSTANCE.createModel
+			createAndRegisterRoot(umlModel, MODEL_NAME.projectModelPath.uri)
+			modelInitialization.apply(umlModel)
+		]
+	}
+
+	protected def void createPcmRepository(String packageName) {
+		changePcmView [
+			val repository = RepositoryFactory.eINSTANCE.createRepository => [
+				it.entityName = packageName
+			]
+			createAndRegisterRoot(repository, MODEL_NAME.projectModelPath.uri)
+		]
+	}
+
+	protected def void createAndRegisterRoot(View view, EObject rootObject, URI persistenceUri) {
+		view.registerRoot(rootObject, persistenceUri)
+	}
+
+	protected def Path getProjectModelPath(String modelName) {
+		Path.of(MODEL_FOLDER_NAME).resolve(modelName + "." + MODEL_FILE_EXTENSION)
+	}
+
+	protected def getDefaultUmlModel(View view) {
+		view.claimUmlModel(MODEL_NAME)
+	}
+
+	protected def getDefaultPcmRepository(View view) {
+		view.claimPcmRepository(PACKAGE_NAME_FIRST_UPPER)
+	}
+
+	def void init_PCM() {
+		userInteraction.addNextTextInput(PcmUmlClassApplicationTestHelper.UML_MODEL_FILE)
+		createPcmRepository(PACKAGE_NAME_FIRST_UPPER)
+	}
+
+	def void init_UML() {
+		userInteraction.addNextSingleSelection(DefaultLiterals.USER_DISAMBIGUATE_REPOSITORY_SYSTEM__REPOSITORY)
+		userInteraction.addNextTextInput(PcmUmlClassApplicationTestHelper.PCM_MODEL_FILE)
+		createUmlRootPackage(PACKAGE_NAME)
+	}
+
+	protected def void changeUmlModel((Model)=>void modelModification) {
+		changeUmlView [
+			modelModification.apply(defaultUmlModel)
+		]
+	}
+
+	protected def void createUmlRootPackage(String packageName) {
+		changeUmlModel [
+			packagedElements += UMLFactory.eINSTANCE.createPackage => [
+				it.name = packageName
+			]
+		]
+	}
+
+	protected def getUmlRootPackage(View view) {
+		view.defaultUmlModel.claimPackage(PACKAGE_NAME)
+	}
+
+	protected def getUmlContractsPackage(View view) {
+		view.defaultUmlModel.getNestedPackages.findFirst[it.name.equals(PACKAGE_NAME)].nestedPackages.findFirst [
+			it.name.equals(CONTRACTS_PACKAGE)
+		]
+	}
+
+	@BeforeEach
+	def void setup() {
+		viewFactory = new PcmUmlclassViewFactory(virtualModel)
+		createUmlModel[name = MODEL_NAME]
+	}
+
+	@AfterEach
+	def protected void cleanup() {
+	}
```

Comment:
If empty, either removed or make abstract if overridden by subclasses.

Reply:
Removed in [c245841](https://github.com/vitruv-tools/Vitruv-CaseStudies/pull/214/commits/c24584179c40c7230b9f788a18f8a912216129cd)